spec/controllers/admin/admin_users_controller_spec.rb

```diff
@@ -547,6 +547,17 @@ def expect_changes_made_by(admin, log_items)
 
               subject.call
             end
+
+            it "deletes marked-as-deleted comments when all its replies are deleted" do
+              reply = create(:comment, commentable: comment, pseud: user.default_pseud)
+              comment.is_deleted = true
+              comment.save!(validate: false)
+
+              subject.call
+
+              expect(Comment.exists?(comment.id)).to be_falsey
+              expect(Comment.exists?(reply.id)).to be_falsey
+            end
```

Comment:
Do you mean something like this?

Reply:
Yep! I think it would be a bit cleaner to call `destroy_or_mark_deleted` on `comment` instead of manually doing the setting and saving (and check that the comment is marked as deleted as expected).